code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -35,14 +35,14 @@
 	var/undies = get_undies()
 	var/gender_socks = get_gender_socks()
 	if(!get_key_by_value(undies, pref.underwear))
-		pref.underwear = undies[undies[1]]
+		pref.underwear = null
 	if(!get_key_by_value(undershirt_t, pref.undershirt))
 		pref.undershirt = undershirt_t[undershirt_t[1]]
 	if(!get_key_by_value(gender_socks, pref.socks))
-		pref.socks = gender_socks[gender_socks[1]]
-
+		pref.socks = null

```

Comment:
Huh, oke, the nulling can be left, though it might still be better to bind this to gender changing as well. Otherwise there might be an issue with changing your gender and it not redrawing proper undies until you edit the equipment itself.

Basically, the call to `sanitize()` shouldn't exist at the start of `content()` in this type. Undies should be handled the moment the gender is swapped.